bento/src/main/java/com/yelp/android/bento/componentcontrollers/RecyclerViewComponentController.java

```diff
@@ -244,6 +245,7 @@ public void clear() {
         mComponentGroup.clear();
         mRecyclerView.removeOnScrollListener(mOnScrollListener);
         mComponentGroup.unregisterComponentDataObserver(mComponentVisibilityListener);
+        mRecyclerViewAdapter.unregisterAdapterDataObserver(mAdapterDataObserver);
```

Comment:
nit: Maybe we should group these in a private method `removeVisibilityListeners()` so that it's more obvious that there's a symmetry where we want to eventually remove whatever we add.

Reply:
Makes sense as well.